### PR TITLE
[ramda] optimize coverage, T, F, startsWith, endsWith 

### PR DESCRIPTION
**Pull request by adispring**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -435,7 +431,7 @@ export function curryN<N extends number>(length: N): <F extends (...args: any) =
 export function dec(n: number): number;
 
 /**
- * Returns the second argument if it is not null or undefined. If it is null or undefined, the
+ * Returns the second argument if it is not null or undefined or NaN. If it is null or undefined or NaN, the
```
Comment (Nicholaiii) [inline types/ramda/index.d.ts:434]: Can we have listing comma? `If it is null, undefined or NaN`

**Review by Nicholaiii — CHANGES_REQUESTED** [review]
As always a great contribution. We could improve legibility with some punctuation.

File: types/ramda/index.d.ts (hunk 1 above)
Comment (adispring) [reply types/ramda/index.d.ts:434]: Ok, I will change it to "," .

Comment (adispring) [reply types/ramda/index.d.ts:434]: How to add NaN as a type ? When I add it after `null | undefined`, Typescript says it is a value, not a type.

Comment (adispring) [reply types/ramda/index.d.ts]: Done.

Comment (Nicholaiii) [reply types/ramda/index.d.ts]: This is unfortunately not possible. There is [an issue for it](https://github.com/Microsoft/TypeScript/issues/28682) but JS doesn't have a NaN type itself: 
image

**Comment by adispring** [comment]
When I optimizing the @types/ramda, I found that the functions' types not unified. some support placeholder and curry, some not. Should we make a plan to normalize all the functions' types?

I found @types/lodash do better than @types/ramda. Maybe we can refer @types/lodash to let all functions support placeholder and curry?

**Comment by Nicholaiii** [comment]
> When I optimizing the @types/ramda, I found that the functions' types not unified. some support placeholder and curry, some not. Should we make a plan to normalize all the functions' types?
> 
> I found @types/lodash do better than @types/ramda. Maybe we can refer @types/lodash to let all functions support placeholder and curry?

Are you referring to the CurriedFunction(n) interfaces they have?

**Comment by adispring** [comment]
No, I have not

**Comment by adispring** [comment]
I am reading lodash .d.ts files recently.

File: types/ramda/index.d.ts (hunk 1 above)
Comment (adispring) [reply types/ramda/index.d.ts]: ok

**Comment by adispring** [comment]
@Nicholaiii Maybe we can merge this pr first :)

**Review by Nicholaiii — APPROVED** [review]

**Comment by adispring** [comment]
Ready to merge

**Pull request merged**